app/services/UserService.scala

```diff
@@ -232,12 +232,47 @@ class UserService @Inject() (
   }
 
   def inactiveActiveFlag(profile: Profile) = {
-    profileDao.setActiveFlag(profile, Some("inactive")).map(_ match{
+    profileDao.setActiveFlag(profile, Some("inactive")).map(_ match {
+      case true => "inactive"
+      case false => None
+    })
```

Comment:
In Scala you always have to consider the result type of a function.

This function looks very awkward, since it tries to return a `String` in one case and an `Option`  (`None`) in another. I'm very confused that your compiler does not cries and cries until it dies... :)
A better way would be:
```
    profileDao.setActiveFlag(profile, Some("inactive")).map(_ match {
      case true => Some("inactive")
      case false => None
    })
```
Thus, your functions result type would be `Option[String]`